File: testdata/cfg_sample.go
package testdata

import (
	"io"
	"net/http"
)

func IfElse(x int) string {
	if x > 0 {
		return "positive"
	} else {
		return "non-positive"
	}
}

func EarlyReturn(r *http.Request) ([]byte, error) {
	body, err := io.ReadAll(r.Body)
	if err != nil {
		return nil, err
	}
	return body, nil
}

func ForLoop(items []string) string {
	var result string
	for _, item := range items {
		result = item
	}
	return result
}

func SwitchCase(op string) int {
	var result int
	switch op {
	case "add":
		result = 1
	case "sub":
		result = 2
	default:
		result = 0
	}
	return result
}

func NestedIfInFor(items []string) int {
	count := 0
	for _, item := range items {
		if len(item) > 3 {
			count++
		}
	}
	return count
}

func HandleValidated(w http.ResponseWriter, r *http.Request) {
	body, err := io.ReadAll(r.Body)
	if err != nil {
		http.Error(w, "bad", 400)
		return
	}
	if len(body) == 0 {
		http.Error(w, "empty", 400)
		return
	}
	_ = body
}

func PanicTerminates(x int) {
	if x < 0 {
		panic("negative")
	}
	_ = x
}

func EmptyFunction() {
}

func LinearFunction() int {
	x := 1
	y := x + 2
	return y
}

File: testdata/cfg_sample.ts
function ifElse(x: number): string {
    if (x > 0) {
        return "positive";
    } else {
        return "non-positive";
    }
}

function earlyReturn(data: string | null): string | null {
    if (!data) {
        return null;
    }
    const result = data.trim();
    return result;
}

function forLoop(items: string[]): string {
    let result = "";
    for (const item of items) {
        result = item;
    }
    return result;
}

function switchCase(op: string): number {
    let result: number;
    switch (op) {
        case "add":
            result = 1;
            break;
        case "sub":
            result = 2;
            break;
        default:
            result = 0;
    }
    return result;
}

function nestedIfInFor(items: string[]): number {
    let count = 0;
    for (const item of items) {
        if (item.length > 3) {
            count++;
        }
    }
    return count;
}

function linearFunction(): number {
    const x = 1;
    const y = x + 2;
    return y;
}

function emptyFunction(): void {
}
